servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3829,9 +3829,31 @@ public void deleteRole(ResourceContext ctx, String domainName, String roleName,
             throw ZMSUtils.requestError("deleteRole: admin role cannot be deleted", caller);
         }
 
+        /* we are not going to allow any user to delete
+         * role that related to any assertion of any policy */
+
+        Policies policies = getPolicies(ctx, domainName, true, true);
+
+        if (policies != null && policies.getList() != null) {
+            validateRoleNotAssociatedToPolicy(policies.getList(), roleName, domainName, caller);
+        }
+
         dbService.executeDeleteRole(ctx, domainName, roleName, auditRef, caller);
     }
 
+    void validateRoleNotAssociatedToPolicy(List<Policy> policies, String roleName, String domainName , String caller) {
+        if (!roleName.startsWith(domainName)) {
```

Comment:
not sure why're doing this check here. this method is only called within deleteRole API and there the user is required to pass the role name only (any other invalid input will be rejected eventually). So we can just generate the roleResourceName without calling startsWith first.

One minor comment as well - remove the extra space after domainName in the signature.